src/main/java/com/wso2telco/logging/PropertyLogHandler.java

```diff
@@ -29,165 +29,126 @@
-    private String getPayloadSting(MessageContext messageContext, org.apache.axis2.context.MessageContext axis2MessageContext) {
-        String payload;
-        if (axis2MessageContext.getProperty(CONTENT_TYPE).equals("application/json")) {
-            /**if content type is json */
-            payload = XML.toJSONObject(messageContext.getEnvelope().getBody().getFirstElement().getFirstElement().toString()).toString();
-        } else if (axis2MessageContext.getProperty(CONTENT_TYPE).equals("text/plain")) {
-            /**if content type is text/plain */
-            payload = messageContext.getEnvelope().getBody().getFirstElement().toString();
-        } else {
-            /** if content type is xml */
-            payload = messageContext.getEnvelope().getBody().toString();
-        }
-        return payload;
-    }
+	private static final String REGISTRY_PATH = "gov:/apimgt/";
+	private static final String MESSAGE_TYPE = "message.type";
+	private static final String PAYLOAD_LOGGING_ENABLED = "payload.logging.enabled";
+	private static final String APPLICATION_ID = "api.ut.application.id";
+	private static final String API_PUBLISHER = "api.ut.apiPublisher";
+	private static final String API_NAME = "API_NAME";
+	private static final String REQUEST = "request";
+	private static final String RESPONSE = "response";
+	private static final String API_VERSION ="SYNAPSE_REST_API_VERSION";
+	private static final String API_CONTEXT = "api.ut.context";
+	private static final String USER_ID="api.ut.userId";
+	private static final String JWT="X-JWT-Assertion";
+	private static final String UUID = "MESSAGE_ID";
+	private static final String ERROR="error";
+	private static final String REQUEST_ID = "mife.prop.requestId";
+	private static final String APPLICATION_NAME = "api.ut.application.name";
+	private static final String  REST_FULL_REQUEST_PATH="REST_FULL_REQUEST_PATH";
+	private static final String  SYNAPSE_REST_API="SYNAPSE_REST_API";
+	private static final String  ERROR_EXCEPTION="ERROR_EXCEPTION";
+	private static final String  API_RESOURCE_CACHE_KEY="API_RESOURCE_CACHE_KEY";
+	private static final String  ERROR_MESSAGE="ERROR_MESSAGE";
+	private static final String  ERROR_CODE="ERROR_CODE";
+	private static final String  HTTP_SC= "HTTP_SC";
+
+	private static final Log logHandler = LogFactory.getLog("REQUEST_RESPONSE_LOGGER");
+
+	public boolean mediate(MessageContext messageContext) {
+
+		boolean isPayloadLoggingEnabled = true;
+
+		org.apache.axis2.context.MessageContext axis2MessageContext = ((Axis2MessageContext) messageContext)
+				.getAxis2MessageContext();
+		isPayloadLoggingEnabled = extractPayloadLoggingStatus(messageContext);
+		String direction = (String) axis2MessageContext.getProperty(MESSAGE_TYPE);
+		if (direction != null && direction.equalsIgnoreCase(REQUEST)) {
+			logRequestProperties(messageContext, axis2MessageContext, isPayloadLoggingEnabled);
+		} else if (direction != null && direction.equalsIgnoreCase(RESPONSE)) {
```

Comment:
You can write the concisely by comparing against the value like
direction != null && direction.equalsIgnoreCase(RESPONSE) -> RESPONSE.equalsIgnoreCase(direction)

Reply:
this was old code, I just changed a boolean value to true. anyway I will have a look on that